File: 프로그래머스/LV2/코딩테스트 연습/월간 코드 챌린지 시즌2/괄호 회전하기/괄호 회전하기.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;

int solution(string s)
{
    int answer = 0;
    int s_len = s.length();
    string temp;
    stack<char> stack;
    s += s;

    for (int i = 0; i < s_len; i++)
    {
        temp = s.substr(i, s_len);
        while (!stack.empty())
            stack.pop();

        for (char c : temp)
        {
            if (!stack.empty() && ((c - stack.top()) == 1 || (c - stack.top()) == 2)) stack.pop();
            else stack.push(c);
        }
        if (stack.empty())answer++;
    }
    return answer;
}

File: 프로그래머스/LV2/코딩테스트 연습/월간 코드 챌린지 시즌2/괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    s_len = len(s)
    s += s
    for i in range(s_len):
        temp = s[i:i+s_len]
        stack = ""
        for c in temp:
            stack += c
            if len(stack) >= 2 and ((ord(stack[-1]) - ord(stack[-2]) == 1) or (ord(stack[-1]) - ord(stack[-2]) == 2)):
                stack = stack[:-2]
        if stack == "":
            answer += 1
    return answer
